pkg/networking/networking.go

```diff
@@ -52,11 +53,50 @@ type defaultHeadersRoundTripper struct {
 	networkAccess            *networkImpl
 }
 
+func (rt *defaultHeadersRoundTripper) logRoundTrip(request *http.Request, response *http.Response, err error) {
+	if rt.networkAccess == nil && rt.networkAccess.logger == nil {
+		return
+	}
+
```

Comment:
Could it be that we need an `||` instead of and `&&`?
https://go.dev/play/p/8WiLadj0lnv